pilot/pkg/networking/core/v1alpha3/gateway_test.go

```diff
@@ -127,7 +131,11 @@ func TestBuildGatewayListenerTlsContext(t *testing.T) {
 					},
 					ValidationContextType: &auth.CommonTlsContext_CombinedValidationContext{
 						CombinedValidationContext: &auth.CommonTlsContext_CombinedCertificateValidationContext{
-							DefaultValidationContext: &auth.CertificateValidationContext{},
+							DefaultValidationContext: &auth.CertificateValidationContext{
+								MatchSubjectAltNames: []*matcher.StringMatcher{
+									{MatchPattern: &matcher.StringMatcher_Prefix{Prefix: spiffe.URIPrefix + spiffe.GetTrustDomain()}},
+								},
+							},
```

Comment:
Add a separate test that shows that when you set the mesh global trust domain aliases, this default validation context gets updated, while this test should show that when the trust domain aliases is not set, then the validation context _does not_ have any trust domain matches (not even the default)

Reply:
>  while this test should show that when the trust domain aliases is not set, then the validation context does not have any trust domain matches (not even the default)

What do you mean by this? Since the user didn't disable td validation at global level, this test shows that the TD from istiod is matched. The test I think you mean we should add is different, when user explicitly sets validation off, this DefaultValidationContext should be empty. That is something I will add

Reply:
See my new comments. The culprit is https://github.com/istio/istio/pull/26224/files#diff-8a26e998c795d3486653dfc885037e28L224 where you are explicitly adding local trust domains (even if the subject alt names at the gateway is empty). This will break the production deployments as I described in my comment. This stems from the fundamental design idea of the doc (turn everything into an explicit allow list whose values are auto populated from local trust domain and mesh config trust domain) and then use authz to select from that subset. This is a big semantics change at the API level- the semantics today is default allow is any trust domain with shared root cert, use authz to restrict.